bundles/AdminBundle/Controller/Admin/Asset/AssetController.php

```diff
@@ -491,7 +492,8 @@ protected function addAsset(Request $request, Config $config)
         if ($context) {
             $context = json_decode($context, true);
             $context = $context ? $context : [];
-            $event = new \Pimcore\Event\Model\Asset\ResolveUploadTargetEvent($parentId, $filename, $context);
+            $context['sourcePath'] = $sourcePath;
+            $event = new ResolveUploadTargetEvent($parentId, $filename, $context);
```

Comment:
I think we could and should do the validation right here, I don't see any benefit of having this in an event ... 🤔

Reply:
@brusch Updated the PR.

Reply:
@storbahn Thanks!